sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -242,6 +246,58 @@ private static synchronized void init(
     notifyOptionsObservers(options);
 
     finalizePreviousSession(options, HubAdapter.getInstance());
+
+    handleStartupProfilingConfig(options, options.getExecutorService());
+  }
+
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private static void handleStartupProfilingConfig(
+      final @NotNull SentryOptions options,
+      final @NotNull ISentryExecutorService sentryExecutorService) {
+    sentryExecutorService.submit(
+        () -> {
+          final String cacheDirPath = options.getCacheDirPathWithoutDsn();
+          if (cacheDirPath != null) {
+            final @NotNull File startupProfilingConfigFile =
+                new File(cacheDirPath, STARTUP_PROFILING_CONFIG_FILE_NAME);
+            // We always delete the config file for startup profiling
+            FileUtils.deleteRecursively(startupProfilingConfigFile);
+            if (!options.isEnableStartupProfiling()) {
+              return;
+            }
+            if (!options.isTracingEnabled()) {
+              options
+                  .getLogger()
+                  .log(
+                      SentryLevel.INFO,
+                      "Tracing is disabled and startup profiling will not start.");
+              return;
+            }
+            try {
+              if (startupProfilingConfigFile.createNewFile()) {
+                final @NotNull TracesSamplingDecision startupSamplingDecision =
+                    sampleStartupProfiling(options);
+                final @NotNull SentryStartupProfilingOptions startupProfilingOptions =
+                    new SentryStartupProfilingOptions(options, startupSamplingDecision);
+                try (Writer fileWriter = new SentryFileWriter(startupProfilingConfigFile)) {
```

Comment:
```suggestion
                try (Writer fileWriter = new FileWriter(startupProfilingConfigFile)) {
```
SentryFileWriter is specific for performance instrumentation, I guess we don't want to create spans for our own file operations :)